lib/core/src/mqtt.rs

```diff
@@ -0,0 +1,122 @@
+use defmt::*;
+#[cfg(not(feature = "std"))]
+use heapless::String;
+use rust_mqtt::{
+    client::{client::MqttClient, client_config::ClientConfig},
+    packet::v5::reason_codes::ReasonCode,
+    utils::rng_generator::CountingRng,
+};
+use serde::{Deserialize, Serialize};
+
+#[cfg(not(feature = "std"))]
+pub struct MqttMessage {
+    pub topic: String<48>,
+    pub payload: String<512>,
+}
+
+#[cfg(feature = "std")]
+pub struct MqttMessage {
+    pub topic: String,
+    pub payload: String,
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct ButtonMqttMessage {
+    pub task_id: u8,
+    pub status: bool,
+}
+
+pub struct HypedMqttClient<
+    'a,
+    T: embedded_io_async::Read + embedded_io_async::Write,
+    R: rand_core::RngCore,
+> {
+    pub client: MqttClient<'a, T, 5, R>,
+}
+
+pub fn initialise_mqtt_config() -> ClientConfig<'static, 5, CountingRng> {
+    let mut config = ClientConfig::new(
+        rust_mqtt::client::client_config::MqttVersion::MQTTv5,
+        CountingRng(20000),
+    );
+    config.add_max_subscribe_qos(rust_mqtt::packet::v5::publish_packet::QualityOfService::QoS1);
+    config.add_client_id("stm-client");
+    config.max_packet_size = 100;
+
+    config
+}
+
+// Implement send_message for HypedMqttClient
+impl<'a, T: embedded_io_async::Read + embedded_io_async::Write, R: rand_core::RngCore>
```

Comment:
Do we want a `new` function here which handles the config rather than the `initialise_mqtt_config` function (means `client` doesn't need to be public too I think)? Could pass in things like the broker address and client ID too

Reply:
Broker address happens at the socket level not the mqtt level so wouldn't work without a lot of borrow checker crying i think but beyond that this has been changed.